### PR TITLE
[ISSUE-604] Proposal CSI Baremetal CI on Github Actions

### LINKED ISSUE #604: CI on Github Actions proposal
Create POC to trigger, run and validate results for CI testing

### PR DESCRIPTION
**Pull request by safronovD**
### Resolves #604

Add proposal

## PR checklist
- [x] Add link to the issue
- [x] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Provide test details_

### CODE REVIEW

**Review by rkiselenko — APPROVED** [review]
LGTM

File: docs/proposals/ci-in-github-action.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+operator_branch=branch1
+```
+
+2. Action: Add a comment to PR
+```
+Start CI. Parameters:
+operator_branch=branch1
+```
+
+3. Action: Perform CI suite
```
Comment (mishoyama) [inline docs/proposals/ci-in-github-action.md:29]: Where do you plan to run CI?

Comment (safronovD) [reply docs/proposals/ci-in-github-action.md:29]: It will be a part of action. I want to show user interactions in this proposal

Comment (mishoyama) [reply docs/proposals/ci-in-github-action.md:29]: I'm asking about target host where you plan to execute CI. How many actions can be executed in parallel? What are the resources available - CPU, Memory, Storage spaces, etc. Do we need to use kind or we can get k8s cluster?

Comment (safronovD) [reply docs/proposals/ci-in-github-action.md:42]: We plan to run CI on Github-hosted runners. They have 7 GB RAM and 15 GB disk space ([reference](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners)). I have found no information about limitations of parallel actions, only in minutes. We need to use kind as we have one host only

Comment (safronovD) [reply docs/proposals/ci-in-github-action.md:42]: Updated info about environment in the proposal

Comment (rkiselenko) [reply docs/proposals/ci-in-github-action.md:42]: Does CI requires `--privileged`  ?

**Review by rkiselenko — CHANGES_REQUESTED** [review]

File: docs/proposals/ci-in-github-action.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+because running tests locally requires much hardware resources and developers' time.
+It is also making simpler to validate Pull Requests from fork-repositories.
+
+## Proposal
+
+We can make our tests running on Github Actions.
+
+Environment:
+- Ubuntu 20
+- Kind 0.11.1
```
Comment (rkiselenko) [inline docs/proposals/ci-in-github-action.md:21]: '--privileged' is needed in order to run e2e CI ?

**Review by rkiselenko — CHANGES_REQUESTED** [review]

File: docs/proposals/ci-in-github-action.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Last updated: 01.11.2021
+
+## Abstract
+
+Description of triggers and user actions to run CSI Baremetal CI.
+
+## Background
+
+We need to create Github Action to perform CSI Baremetal CI tests, 
+because running tests locally requires much hardware resources and developers' time.
```
Comment (mishoyama) [inline docs/proposals/ci-in-github-action.md:12]: What do you mean by `locally`?

File: docs/proposals/ci-in-github-action.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+5. Create separate Github Action in CSI Baremetal Operator repo (ETA - 1)
+6. Investigate to the way to create new repo for e2e tests with new information about actions usability (ETA - 1)
+
+## Open issues (if applicable)
+
+ID | Name | Descriptions | Status | Comments
+---| -----| -------------| ------ | --------
+ISSUE-1 | Script for CI  | We need to have script for CI testing  |   |
+ISSUE-2 | Parsing CI results  | It is better to have a human readable report with info about number of passed/failed/skipped tests  |   |
+ISSUE-3 | Should we have separate repo for e2e tests?  |   |   |
```
Comment (mishoyama) [inline docs/proposals/ci-in-github-action.md:76]: Could you please add issue for time constraints?

File: docs/proposals/ci-in-github-action.md (hunk 2 above)
Comment (safronovD) [reply docs/proposals/ci-in-github-action.md:21]: No, it works well even for LVG volumes. But I don't know how...

File: docs/proposals/ci-in-github-action.md (hunk 3 above)
Comment (safronovD) [reply docs/proposals/ci-in-github-action.md]: changed with `developer's environment`

File: docs/proposals/ci-in-github-action.md (hunk 4 above)
Comment (safronovD) [reply docs/proposals/ci-in-github-action.md:76]: add issue about it

**Review by mishoyama — APPROVED** [review]

**Review by rkiselenko — APPROVED** [review]

**Pull request merged**